polar-core/tests/integration_tests.rs

```diff
@@ -1569,7 +1569,12 @@ fn test_rest_vars() -> TestResult {
     qeval(&mut p, "not append([1,2,3], [4], [1,2,3])");
 
     let a = &var(&mut p, "[*_] in [*a] and [*b] in [*_] and b = 1", "a")[0];
-    assert!(matches!(a, Value::List(b) if matches!(b[0].value(), Value::RestVariable(_))));
+    assert!(matches!(a,
```

Comment:
like I mentioned in the other PR, I kept this test but I'm not sure what it's supposed to be testing.

Reply:
https://github.com/osohq/oso/pull/644

Reply:
Feel free to replace with a clearer test. Agree that this test isn't very easy to follow!

Reply:
List in list doesn't make much sense here.

Maybe changing them to unifies? `[*_] = [*a] and [*b] = [*_] and b = [1]` ?

Reply:
can I replace it with a test that just ensures `a` and `b` get different bindings? the specific outcome we were/are testing for doesn't seem "correct" as much as just "not specifically incorrect".